code/cbr.py

```diff
@@ -390,6 +390,22 @@ def do_symbolic_case_based_reasoning(self):
                 for rel, _ in learnt_programs[k].items():
                     logger.info((rel, learnt_programs[k][rel]))
                 logger.info("=====" * 2)
+        if args.parallelize:
+            dirname = os.path.join(self.args.output_dir,\
+                                   self.args.dataset_name)
+            if not os.path.exists(dirname):
+                os.makedirs(dirname)
+
```

Comment:
Here, you should make sure that any existing files in the directory are removed. So that the aggregate script does not over count results from previous runs.